File: herencia-dto/SergioDeIscar-herencia-java/SergioDeIscar-herencia-java/src/main/java/org/example/models/Alumno.java
package org.example.models;

import org.example.dto.PersonaDto;

import java.io.Serializable;

public class Alumno extends Persona implements Serializable {
    private final int edad;

    public Alumno(String nombre, int edad) {
        super(nombre);
        this.edad = edad;
    }

    public Alumno(Integer id, String nombre, int edad) {
        super(id, nombre);
        this.edad = edad;
    }

    @Override
    public String toString() {
        return "Alumno{" +
                "id=" + getId() +
                ", nombre='" + getNombre() + '\'' +
                ", edad=" + edad +
                '}';
    }

    public int getEdad() {
        return edad;
    }

    @Override
    public PersonaDto toDto() {
        return new PersonaDto(
                Integer.toString(getId()),
                getNombre(),
                "Alumno",
                Integer.toString(getEdad()),
                "null"
        );
    }
}

File: herencia-dto/SergioDeIscar-herencia-kotlin/SergioDeIscar-herencia-kotlin/src/main/kotlin/models/Alumno.kt
package models

import dto.PersonaDto
import java.io.Serializable

class Alumno(
    id: Long = 0,
    nombre: String,
    val edad: Int
) : Persona(id, nombre), Serializable {

    override fun toCsvRow(): String {
        return "$id,$nombre,Alumno,$edad,null\n"
    }

    override fun copy(id: Long, nombre: String): Persona {
        this.id = id
        return Alumno(id, nombre, edad)
    }

    override fun toString(): String {
        return "Alumno(id=$id, nombre='$nombre', edad=$edad)"
    }
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Alumno) return false
        return id == other.id && nombre == other.nombre && edad == other.edad
    }
}